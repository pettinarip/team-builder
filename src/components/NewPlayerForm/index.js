import NewPlayerForm from './NewPlayerForm'

export default NewPlayerForm
