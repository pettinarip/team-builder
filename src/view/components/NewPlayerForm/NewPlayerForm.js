import React from 'react'
import { Field } from 'redux-form'
import PropTypes from 'prop-types'
import SettingField from '../SettingField'

const required = value => (value ? undefined : 'Required')
const between = (min, max) =>
  (value, previousValue) => value <= max && value >= min ? value : previousValue
const maxLength = max =>
  (value, previousValue) => value.length <= max ? value : previousValue

const NewPlayerForm = ({ handleSubmit, pristine, submitting }) => (
  <form className='form-horizontal' onSubmit={handleSubmit}>
    <SettingField title='Name'>
      <Field
        autoFocus
        name='name'
        component='input'
        type='text'
        placeholder='Francesco Totti'
        validate={required}
        normalize={maxLength(30)}
      />
    </SettingField>
    <SettingField title='Number'>
      <Field
        name='number'
        component='input'
        type='number'
        placeholder='10'
        validate={required}
        normalize={between(1, 99)}
      />
    </SettingField>
    <SettingField>
      <button className='primary-button save-button button' type='submit' disabled={pristine || submitting}>
        Add to the team
      </button>
    </SettingField>
  </form>
)

NewPlayerForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool
}

export default NewPlayerForm
