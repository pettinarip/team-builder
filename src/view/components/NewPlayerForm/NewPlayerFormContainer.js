import { reduxForm } from 'redux-form'
import NewPlayerForm from './NewPlayerForm'

export default reduxForm({
  form: 'newPlayer'
})(NewPlayerForm)
