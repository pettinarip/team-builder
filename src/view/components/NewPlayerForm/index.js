import NewPlayerFormContainer from './NewPlayerFormContainer'

export default NewPlayerFormContainer
