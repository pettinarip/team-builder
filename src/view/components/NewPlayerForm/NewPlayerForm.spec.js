import React from 'react'
import { shallow } from 'enzyme'
import NewPlayerForm from './NewPlayerForm'

describe('NewPlayerForm component', () => {
  it('should render the component properly', () => {
    const component = mountForm()
    expect(component).toMatchSnapshot()
  })

  it('should call handleSubmit on submit', () => {
    const handleSubmit = jest.fn()
    const component = mountForm({
      handleSubmit
    })

    component.find('form').simulate('submit')
    expect(handleSubmit).toHaveBeenCalled()
  })
})

function mountForm (props) {
  const propsToUse = {
    handleSubmit () {},
    ...props
  }

  return shallow(<NewPlayerForm {...propsToUse} />)
}
