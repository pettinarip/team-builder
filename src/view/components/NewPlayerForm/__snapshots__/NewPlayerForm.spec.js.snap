// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPlayerForm component should render the component properly 1`] = `
<form
  className="form-horizontal"
  onSubmit={[Function]}
>
  <SettingField
    title="Name"
  >
    <Field
      autoFocus={true}
      component="input"
      name="name"
      normalize={[Function]}
      placeholder="Francesco Totti"
      type="text"
      validate={[Function]}
    />
  </SettingField>
  <SettingField
    title="Number"
  >
    <Field
      component="input"
      name="number"
      normalize={[Function]}
      placeholder="10"
      type="number"
      validate={[Function]}
    />
  </SettingField>
  <SettingField>
    <button
      className="primary-button save-button button"
      type="submit"
    >
      Add to the team
    </button>
  </SettingField>
</form>
`;
